packages/block-library/src/terms-query/inspector-controls.js

```diff
@@ -181,6 +182,33 @@ export default function TermsQueryInspectorControls( {
 							disabled={ termQuery.parent === 0 }
 						/>
 					</ToolsPanelItem>
+
+					{ ! termQuery.hierarchical && (
+						<ToolsPanelItem
+							hasValue={ () => termQuery.perPage !== 10 }
+							label={ __( 'Max terms' ) }
+							onDeselect={ () => setQuery( { perPage: 10 } ) }
+							isShownByDefault
+						>
+							<RangeControl
+								__nextHasNoMarginBottom
+								__next40pxDefaultSize
+								label={ __( 'Max terms' ) }
+								value={ termQuery.perPage }
+								min={ 0 }
+								max={ 100 }
+								onChange={ ( perPage ) => {
+									// Show all terms (-1) when 0 is selected.
+									setQuery( {
+										perPage: perPage === 0 ? -1 : perPage,
+									} );
+								} }
+								help={ __(
+									'Limit the terms you want to show. To show all terms, use 0 (zero).'
```

Comment:
Very tiny wording suggestion:

```suggestion
									'Limit the number of terms you want to show. To show all terms, use 0 (zero).'
```

Reply:
Done https://github.com/WordPress/gutenberg/pull/71358/commits/d5ec43263b34d95c21ced7fdb2695b8b3b9b031c